hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -154,15 +156,29 @@ public BackgroundTaskResult call() {
             subFileNum += request.getDeletedSubFilesCount();
           }
 
-          optimizeDirDeletesAndSubmitRequest(remainNum, dirNum, subDirNum,
-              subFileNum, allSubDirList, purgePathRequestList, null, startTime);
+          // Acquire deletedTable write lock this late to allow KeyDeletingTask
+          // to interleave up until this point
+          getOzoneManager().getMetadataManager().getTableLock(
```

Comment:
Do we need `DELETED_TABLE` lock here? We are not accessing `deletedTable` in the `DirectoryDeletingService`

Reply:
I think you are right.

Initially I noticed `deletedTable` write inside `OMDirectoriesPurgeResponseWithFSO#addToDBBatch`. But `addToDBBatch()`es are serialized so that's not an actual concern. Because `createOmSnapshotCheckpoint()` is called inside another `addToDBBatch()` as well. Especially now that we solved the `deletedTable` race condition with #4602 .